src/DetailsView/components/issue-filing-dialog.tsx

```diff
@@ -1,77 +1,138 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+import { cloneDeep, isEqual } from 'lodash';
 import { Dialog, DialogFooter, DialogType } from 'office-ui-fabric-react/lib/Dialog';
 import * as React from 'react';
 
-import { BugFilingSettingsContainer, BugFilingSettingsContainerDeps } from '../../bug-filing/components/bug-filing-settings-container';
+import { BugFilingServiceProvider } from '../../bug-filing/bug-filing-service-provider';
+import {
+    BugFilingSettingsContainer,
+    BugFilingSettingsContainerDeps,
+    OnPropertyUpdateCallback,
+    OnSelectedServiceChange,
+} from '../../bug-filing/components/bug-filing-settings-container';
 import { BugFilingService } from '../../bug-filing/types/bug-filing-service';
 import { EnvironmentInfoProvider } from '../../common/environment-info-provider';
-import { NamedSFC } from '../../common/react/named-sfc';
+import { UserConfigMessageCreator } from '../../common/message-creators/user-config-message-creator';
 import { CreateIssueDetailsTextData } from '../../common/types/create-issue-details-text-data';
-import { BugServiceProperties } from '../../common/types/store-data/user-configuration-store';
+import { BugServicePropertiesMap } from '../../common/types/store-data/user-configuration-store';
 import { ActionAndCancelButtonsComponent } from './action-and-cancel-buttons-component';
 
 export interface IssueFilingDialogProps {
     deps: IssueFilingDialogDeps;
     isOpen: boolean;
     selectedBugFilingService: BugFilingService;
     selectedBugData: CreateIssueDetailsTextData;
-    selectedBugFilingServiceData: BugServiceProperties;
     bugFileTelemetryCallback: (ev: React.SyntheticEvent) => void;
+    bugServicePropertiesMap: BugServicePropertiesMap;
     onClose: (ev: React.SyntheticEvent) => void;
 }
 
 export type IssueFilingDialogDeps = {
     environmentInfoProvider: EnvironmentInfoProvider;
+    userConfigMessageCreator: UserConfigMessageCreator;
+    bugFilingServiceProvider: BugFilingServiceProvider;
 } & BugFilingSettingsContainerDeps;
 
 const titleLabel = 'Specify issue filing location';
 
-export const IssueFilingDialog = NamedSFC<IssueFilingDialogProps>('IssueFilingDialog', props => {
-    const onPrimaryButtonClick = (ev: React.SyntheticEvent<Element, Event>) => {
-        props.bugFileTelemetryCallback(ev);
-        props.onClose(ev);
-    };
+export interface IssueFilingDialogState {
+    selectedBugFilingService: BugFilingService;
```

Comment:
I'm wondering if it is better to only save the service key in the state instead of the entire object.

Reply:
I went with service because it would result in fewer changes on the currently file. I can re-factor it in a future PR.

Reply:
Sounds good. this PR is already huge...